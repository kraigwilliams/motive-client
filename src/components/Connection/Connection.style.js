import styled from 'styled-components'
import { colors } from '../constants'

export const ConnectionDiv = styled.div`
  
  color: ${colors.offwhite};
  display: flex;
  flex-direction: row;
  text-align: center;
  align-self: center;
  justify-content: center;
  border-radius: 10px;
  max-width: 100%;
  padding-top: 20px;
  padding-bottom: 20px;
`;

export const ConnectionName = styled.p`
  color: ${colors.offwhite};
  text-decoration: none;
  line-height: 2.5;
  border-bottom: 1px solid ${colors.darkergrey};
  display: flex;

  .connection-details{
    display: flex;
    font-size: 14px;
    align-items: center;
    color: ${colors.blue};
  }

  .connection-fullname {
    display: flex;
    font-size: 16px;
    margin-right: 10px;
    color: ${colors.offwhite};
  }

  @media (min-width: 600px) {
    text-align: center;
    font-size: 25px;
  } 
`;