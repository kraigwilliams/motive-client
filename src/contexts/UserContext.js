import React, { Component } from "react";
import AuthApiService from "../services/auth-api-service";
import TokenService from "../services/token-service";
import IdleService from "../services/idle-service";

const UserContext = React.createContext({
  user: {},
  error: null,
  addedConnection: null,
  setError: () => {},
  clearError: () => {},
  setUser: () => {},
  processLogin: () => {},
  processLogout: () => {},
  setAddedConnection: () => {},
});

export default UserContext;

export class UserProvider extends Component {
  constructor(props) {
    super(props);
    this.state = {
      user: {},
      error: null,
      addedConnection: null,
    };

    const jwtPayload = TokenService.parseAuthToken();

    if (jwtPayload) {
      this.state.user = {
        id: jwtPayload.user_id,
        firstname: jwtPayload.firstname,
        lastname: jwtPayload.lastname,
        username: jwtPayload.sub,
      };

      IdleService.setIdleCallback(this.logoutBecauseIdle);
    }
  }

  componentDidMount() {
    if (TokenService.hasAuthToken()) {
      IdleService.registerIdleTimerResets();
      TokenService.queueCallbackBeforeExpiry(() => {
        this.fetchRefreshToken();
      });
    }
  }

  componentWillUnmount() {
    IdleService.unRegisterIdleResets();
    TokenService.clearCallbackBeforeExpiry();
  }

  setError = (error) => {
    this.setState({ error });
  };

  clearError = () => {
    this.setState({ error: null });
  };

  setUser = (user) => {
    this.setState({ user });
  };

  processLogin = (authToken) => {
    TokenService.saveAuthToken(authToken);
    const jwtPayload = TokenService.parseAuthToken();
    this.setUser({
      id: jwtPayload.user_id,
      firstname: jwtPayload.firstname,
      lastname: jwtPayload.lastname,
      username: jwtPayload.sub,
    });
    IdleService.registerIdleTimerResets();
    TokenService.queueCallbackBeforeExpiry(() => {
      this.fetchRefreshToken();
    });
  };

  processLogout = () => {
    TokenService.clearAuthToken();
    TokenService.clearCallbackBeforeExpiry();
    IdleService.unRegisterIdleResets();
    this.setUser({});
  };

  logoutBecauseIdle = () => {
    TokenService.clearAuthToken();
    TokenService.clearCallbackBeforeExpiry();
    IdleService.unRegisterIdleResets();
    this.setUser({ idle: true });
  };

  fetchRefreshToken = () => {
    AuthApiService.refreshToken()
      .then((res) => {
        TokenService.saveAuthToken(res.authToken);
        TokenService.queueCallbackBeforeExpiry(() => {
          this.fetchRefreshToken();
        });
      })
      .catch((err) => {
        this.setError(err);
      });
  };

  setAddedConnection = (value) => {
    this.setState({
      addedConnection: value,
    });
  };

  render() {
    const value = {
      user: this.state.user,
      error: this.state.error,
      addedConnection: this.state.addedConnection,
      setError: this.setError,
      clearError: this.clearError,
      setUser: this.setUser,
      processLogin: this.processLogin,
      processLogout: this.processLogout,
      setAddedConnection: this.setAddedConnection,
    };
    return (
      <UserContext.Provider value={value}>
        {this.props.children}
      </UserContext.Provider>
    );
  }
}
