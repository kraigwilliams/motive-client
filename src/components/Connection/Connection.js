import React, { Component } from 'react'
import UserContext from '../../contexts/UserContext'
import { ConnectionName, ConnectionDiv } from './Connection.style'
import { AddConnection } from '../Button/Button';
import ActionsService from '../../services/actions-service'


class Connection extends Component {
  static contextType = UserContext;

  constructor(props) {
    super(props)
    this.state = {
      userId: null,
    }
  }

  componentDidMount() {
    const { user } = this.context;
    const userId = user.id;

    this.setState({
      userId
    })
  }

  async handleAddConnection() {
    const connectionId = this.props.id
    await ActionsService.addConnection(this.state.userId, connectionId)

    // ActionsService.getConnections(this.state.userId)
  }


  render() {
    return (
      <ConnectionDiv key={this.props.id}>
        <ConnectionName>
          <AddConnection 
            marginleft='0px' 
            marginright='10px' 
            type='button'  
            onClick={this.handleAddConnection}
          />
          {this.props.firstname}
          {this.props.lastname}
          {this.props.username}
        </ConnectionName>
      </ConnectionDiv>
    );
  }
}

export default Connection;