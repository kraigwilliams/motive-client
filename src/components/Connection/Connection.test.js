import React from "react";
import { shallow, configure } from "enzyme";
import toJson from "enzyme-to-json";
import Connection from "./Connection";
import Adapter from "enzyme-adapter-react-16";
import UserContext from "../../contexts/UserContext";

configure({ adapter: new Adapter() });
describe(`<Connection />`, () => {
  //Snapshot Testing
  it("renders Connection by default", () => {
    const wrapper = shallow(
      <UserContext.Provider>
        <Connection />
      </UserContext.Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
