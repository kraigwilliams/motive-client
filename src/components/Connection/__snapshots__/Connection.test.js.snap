// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Connection /> renders Connection by default 1`] = `<Connection />`;
